### PR TITLE
Add arm64 source tests

### PR DESCRIPTION
**Pull request by albertvaka**
### What does this PR do?

- Bump the build images version to include python3.8 on the arm builders (the previous ones used the system python, which is too old to run our tests).
- Add source tests that run on arm64.

### Motivation

Recently we had an arm-only issue where it was discussed we should add tests: https://github.com/DataDog/datadog-agent/pull/7125

### CODE REVIEW

File: .gitlab/source_test/linux.yml [hunk 1]
```diff
@@ -7,7 +7,6 @@
   rm -f vendor.tar.gz
 
 .linux_tests_preparation:
```
Comment (mx-psi) [inline .gitlab/source_test/linux.yml:9]: We could add something like:
```yaml
script:
  - inv -e test --race --profile --python-runtimes "$PYTHON_RUNTIMES" --cpus 4 $EXTRA_OPTS
```

to `.linux_tests_preparation` and set `$EXTRA_OPTS` to `--build-exclude=systemd`when necessary; that way we avoid some repetition.

**Review by mx-psi — COMMENTED** [review]
LGTM, just a minor comment

File: .gitlab/source_test/linux.yml [hunk 2]
```diff
@@ -17,56 +17,87 @@
...
+  needs: ["linux_arm64_go_deps"]
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/deb_arm64:$DATADOG_AGENT_BUILDIMAGES
+  tags: ["runner:docker-arm", "platform:arm64"]
+  variables:
+    PYTHON_RUNTIMES: '3'
+    CONDA_ENV: ddpy3
+
+tests_rpm-arm64-py2:
+  extends: .linux_tests
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/rpm_arm64:$DATADOG_AGENT_BUILDIMAGES
```
Comment (bkabrda) [inline .gitlab/source_test/linux.yml:87]: Did you forget to specify `needs` here as you do in the other tests?

**Review by mx-psi — APPROVED** [review]
LGTM after addressing @bkabrda's comment

File: .gitlab/source_test/linux.yml (hunk 2 above)
Comment (albertvaka) [reply .gitlab/source_test/linux.yml:88]: Yes I did, should be fixed now :)

**Pull request merged**